internal/api/provider/apple.go

```diff
@@ -45,38 +31,33 @@ type appleUser struct {
 	Email string    `json:"email"`
 }
 
-type idTokenClaims struct {
-	jwt.StandardClaims
-	AccessTokenHash string `json:"at_hash"`
-	AuthTime        int    `json:"auth_time"`
-	Email           string `json:"email"`
-	IsPrivateEmail  bool   `json:"is_private_email,string"`
-	Sub             string `json:"sub"`
-}
-
 // NewAppleProvider creates a Apple account provider.
-func NewAppleProvider(ext conf.OAuthProviderConfiguration) (OAuthProvider, error) {
+func NewAppleProvider(ctx context.Context, ext conf.OAuthProviderConfiguration) (OAuthProvider, error) {
 	if err := ext.Validate(); err != nil {
 		return nil, err
 	}
 
-	authHost := chooseHost(ext.URL, defaultAppleAPIBase)
+	if ext.URL != "" {
+		logrus.Warn("Apple OAuth provider has URL config set which is ignored (check GOTRUE_EXTERNAL_APPLE_URL)")
+	}
+
+	oidcProvider, err := oidc.NewProvider(ctx, IssuerApple)
+	if err != nil {
+		return nil, err
+	}
 
 	return &AppleProvider{
 		Config: &oauth2.Config{
-			ClientID:     ext.ClientID,
+			ClientID:     ext.ClientID[0],
 			ClientSecret: ext.Secret,
-			Endpoint: oauth2.Endpoint{
-				AuthURL:  authHost + authEndpoint,
-				TokenURL: authHost + tokenEndpoint,
-			},
+			Endpoint:     oidcProvider.Endpoint(),
 			Scopes: []string{
-				scopeEmail,
-				scopeName,
+				"email",
+				"name",
```

Comment:
i think this might be `fullName` instead of `name` based on (https://developer.apple.com/documentation/authenticationservices/asauthorization/scope)